File: problems/0100-0200/0136_single-number/solution_xor.cpp
class Solution {
  public:
    int singleNumber(vector<int> &nums) {
        // XOR fold: x ^ x = 0 cancels each pair, x ^ 0 = x passes the lone
        // value through, and commutativity makes grouping order irrelevant.
        int result = 0;
        for (int value : nums) {
            result ^= value;
        }
        // Only the unpaired element survives in the accumulator.
        return result;
    }
};

File: problems-adapt/0101-0200/0136_lone-element/solution_xor.py
class Solution:
    def loneElement(self, nums: list[int]) -> int:
        # XOR fold: x ^ x = 0 cancels each pair, x ^ 0 = x passes the lone
        # value through, and commutativity makes grouping order irrelevant.
        result = 0
        for value in nums:
            result ^= value
        # Only the unpaired element survives in the accumulator.
        return result
